pkgs/stdenv/generic/setup.sh

```diff
@@ -843,9 +863,12 @@ _defaultUnpack() {
 
 
 unpackFile() {
+    local status
     curSrc="$1"
     header "unpacking source archive $curSrc" 3
-    if ! runOneHook unpackCmd "$curSrc"; then
+    runOneHook unpackCmd "$curSrc"
+    status=$?
+    if (( status != 0)); then
```

Comment:
We don't need to capture the exit code. We do nothing with it.

Reply:
So the problem here is that the conditional is a little complicated, osh doesn't like it.
I thought we would just run the command and check the status in a separate conditional.

Reply:
Here is an additonal thought, the semantics here have changed a little. If the unpack fails, this will exit on errexit.
I'm thinking that is what we want.